src/sentry/static/sentry/app/components/dropdownControl.jsx

```diff
@@ -0,0 +1,108 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import styled from 'react-emotion';
+import DropdownButton from 'app/components/dropdownButton';
+import DropdownMenu from 'app/components/dropdownMenu';
+import space from 'app/styles/space';
+
+/*
+ * A higher level dropdown component that helps with building complete dropdowns
+ * including the button + menu options. Use the `button` or `label` prop to set
+ * the button content and `children` to provide menu options.
+ */
+class DropdownControl extends React.Component {
+  static propTypes = {
+    // String or element for the button contents.
+    label: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+    // A closure that returns a styled button. Function will get (isOpen, getActorProps)
+    // as arguments. Use this if you need to style/replace the dropdown button.
+    button: PropTypes.func,
+    // Width of the menu. Defaults to 100% of the button width.
+    menuWidth: PropTypes.string,
+    // Height offset for the menu. Defaults to 39px as standard buttons are
+    // 40px tall
+    menuOffset: PropTypes.string,
+    // Should the menu contents always be rendered?  Defaults to true.
+    // Set to false to have menu contents removed from the DOM on close.
+    alwaysRenderMenu: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    menuOffset: '39px',
+    alwaysRenderMenu: true,
+    menuWidth: '100%',
+  };
+
+  renderButton(isOpen, getActorProps) {
+    const {label, button} = this.props;
+    if (button) {
+      return button(isOpen, getActorProps);
```

Comment:
Maybe make this an object to be consistent with DropdownMenu render prop.

Reply:
Good idea :+1: